### PR TITLE
Add `path` and `project_name` to OTEL logs

### PR DESCRIPTION
**Pull request by Cloud11PL**
## Scope of the PR

* adds `path` and `project_name` to OTEL logs

## Checklist

- [ ] I added changesets and [read good practices](/.changeset/README.md).

### CODE REVIEW

File: apps/avatax/src/logger-context.ts [hunk 1]
```diff
@@ -3,4 +3,4 @@ import { LoggerContext } from "@saleor/apps-logger/node";
 /**
  * Server-side only
  */
-export const loggerContext = new LoggerContext();
+export const loggerContext = new LoggerContext("saleor-app-avatax");
```
Comment (lkostrowski) [inline apps/avatax/src/logger-context.ts:6]: we should pass this name from env etc. Maybe there is Vercel env with project name? If not, create one

File: packages/logger/src/logger-context.ts [hunk 2]
```diff
@@ -18,7 +20,13 @@ export class LoggerContext {
   }
 
   async wrap(fn: (...args: unknown[]) => unknown, initialState = {}) {
-    return this.als.run(initialState, fn);
+    return this.als.run(
+      {
+        ...initialState,
+        project_name: this.projectName,
```
Comment (lkostrowski) [inline packages/logger/src/logger-context.ts:26]: also add `service`. In otel and datadog `service` is the right name. `project_name` is extra for vercel

**Review by lkostrowski — CHANGES_REQUESTED** [review]

File: packages/logger/src/logger-context.ts (hunk 2 above)
Comment (lkostrowski) [reply packages/logger/src/logger-context.ts:26]: ah good, thats fine

File: apps/avatax/src/logger-context.ts (hunk 1 above)
Comment (lkostrowski) [reply apps/avatax/src/logger-context.ts:6]: So maybe we should pass here `OTEL_SERVICE_NAME` too? 🤔

Comment (Cloud11PL) [reply apps/avatax/src/logger-context.ts:6]: For consistency I think so

**Review by lkostrowski — APPROVED** [review]

File: packages/logger/src/logger-context.ts [hunk 3]
```diff
@@ -4,6 +4,8 @@ import { SALEOR_API_URL_HEADER, SALEOR_EVENT_HEADER } from "@saleor/app-sdk/cons
 
 export class LoggerContext {
   private als = new AsyncLocalStorage<Record<string, unknown>>();
+  // eslint-disable-next-line turbo/no-undeclared-env-vars
```
Comment (krzysztofzuraw) [inline packages/logger/src/logger-context.ts:7]: Don't we need to add it to turbo.json for other apps like we have in avatax?

**Review by krzysztofzuraw — CHANGES_REQUESTED** [review]

File: packages/logger/src/logger-context.ts (hunk 3 above)
Comment (Cloud11PL) [reply packages/logger/src/logger-context.ts:7]: Not sure, but it's been like that for a good while 😛 
https://github.com/saleor/apps/blob/main/packages/logger/src/logger-otel-transport.ts#L58

Comment (krzysztofzuraw) [reply packages/logger/src/logger-context.ts:7]: I suggest we try to add it into turbo.json where needed and see 😄 - it will help us with turbo cache etc.

Comment (lkostrowski) [reply packages/logger/src/logger-context.ts:7]: We need to test it, I think there is no workspace turbo json file for packages, not sure if linter can handle this anyway. So we should probably have this env in root turbo config, and probably keep the comment

Comment (Cloud11PL) [reply packages/logger/src/logger-context.ts:7]: Yeah, linter doesn't handle it at all (very strange) - the env is in avatax and products-feed. We need to keep it as it is

Comment (lkostrowski) [reply packages/logger/src/logger-context.ts:7]: Its not strage, turbo.json is in apps only, not in packages

Comment (krzysztofzuraw) [reply packages/logger/src/logger-context.ts:7]: @Cloud11PL I see you added env var to turbo.json so this comment can be removed?

Comment (Cloud11PL) [reply packages/logger/src/logger-context.ts:7]: Even with env var added linter still shows an error 😅

**Review by poulch — APPROVED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Pull request merged**